lib/facebook/messenger/bot.rb

```diff
@@ -18,8 +18,15 @@ class << self
         #
         # Returns a String describing the message ID if the message was sent,
         # or raises an exception if it was not.
-        def deliver(message)
-          response = post '/messages', body: JSON.dump(message), format: :json
+        def deliver(message, options = {})
```

Comment:
I just realised that you did this to be backwards compatible, but I'm leaning towards making a breaking change now. The two are just so radically different that it's painful to support both, and I suspect users would be confused to find that you have to pass a hash with `access_token` in it as an argument to everything _but only if_ you've configured the gem with a provider.